src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright @ 2020-Present 8x8, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.videobridge.message
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter
+import com.fasterxml.jackson.annotation.JsonAnySetter
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.JsonMappingException
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+
+/**
+ * Represent a message sent over the "bridge channel" between an endpoint (or "client") and jitsi-videobridge, or
+ * between two jitsi-videobridge instances over Octo.
+ *
+ * The messages are formatted in JSON with a required "colibriClass" field, which indicates the message type. Different
+ * message types have different (if any) additional fields.
+ */
+sealed class BridgeChannelMessage(
+    @get:JsonProperty("colibriClass")
+    val type: String
+) {
+    fun toJson(): String = ObjectMapper().writeValueAsString(this)
+
+    companion object {
+        @JsonIgnoreProperties(ignoreUnknown = true)
+        private class EmptyBridgeChannelMessage : BridgeChannelMessage("")
+
+        @Throws(InvalidMessageTypeException::class, JsonProcessingException::class, JsonMappingException::class)
+        fun parse(string: String): BridgeChannelMessage {
+            // First parse as an empty message, ignoring any fields other than colibriClass, in order to get the type.
+            val colibriClass = jacksonObjectMapper().readValue<EmptyBridgeChannelMessage>(string).type
```

Comment:
I think I can see why you didn't do a custom deserializer (looking at some examples, it looks like it would be overkill unless there's a good way to leverage existing code after doing the first bit), but I found another strategy that might fit better [here](https://stackoverflow.com/a/44123978/2734014).  This way is a bit confusing since creating an `EmptyBridgeChannelMessage` instantiates the parent `type` field with an empty string, but somehow it does magically get filled out--to be honest I'm not even sure how that field gets properly set considering it isn't even an argument to `EmptyBridgeChannelMessage`.